src/Connect/core_interface.cpp

```diff
@@ -121,16 +121,16 @@ device_params_t core_interface::get_data() {
         switch (marlin_vars->print_state) {
```

Comment:
Maybe split this function into two? Separate the marlin->connect state mapping from copying of variables.